lib/services/livesync/android-livesync-service.ts

```diff
@@ -35,7 +35,17 @@ class AndroidLiveSyncService extends liveSyncServiceBaseLib.LiveSyncServiceBase<
 
 	public beforeLiveSyncAction(deviceAppData: Mobile.IDeviceAppData): IFuture<void> {
 		return (() => {
-			let deviceRootPath = this.getDeviceRootPath(deviceAppData.appIdentifier);
+			let deviceRootPath = this.getDeviceRootPath(deviceAppData.appIdentifier),
+				deviceRootDir = path.dirname(deviceRootPath),
+				deviceRootBasename = path.basename(deviceRootPath),
+				listResult = this.device.adb.executeShellCommand(["ls", "-l", deviceRootDir]).wait(),
+				regex = new RegExp(`^-.*${deviceRootBasename}$`, "m"),
+				matchingFile = (listResult || "").match(regex);
+
+			if(matchingFile && matchingFile[0] && _.startsWith(matchingFile[0], '-')){
+				this.device.adb.executeShellCommand(["rm", "-rf", deviceRootPath]).wait();
```

Comment:
I would love to see some `log.trace` message here, informing that the file had been removed